src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
```

Comment:
If you are okay with it then the last style (the "sufficient" one) would be nice :) that way we have one way to get the data that is correct and won't end up having to support more things when we refactor/change stuff.